torchmetrics/metric.py

```diff
@@ -180,6 +180,10 @@ def forward(self, *args, **kwargs):
 
     def _sync_dist(self, dist_sync_fn=gather_all_tensors):
         input_dict = {attr: getattr(self, attr) for attr in self._reductions.keys()}
+        for attr, reduction_fn in self._reductions.items():
+            # dim_zero_cat can be applied multiple times, this optimizes the number of gathers
```

Comment:
just personal preference
```suggestion
            # pre-concatenate metric states that are lists to reduce number of all_gather operations
```